deepchem/models/torch_models/grover.py

```diff
@@ -92,3 +98,312 @@ def forward(self, graph_batch: List[torch.Tensor]):
+        sigmoid = nn.Sigmoid()
+
+        av_atom_loss, av_bond_loss, av_dist_loss = 0.0, 0.0, 0.0
+        fg_atom_from_atom_loss, fg_atom_from_bond_loss, fg_atom_dist_loss = 0.0, 0.0, 0.0
+        bv_atom_loss, bv_bond_loss, bv_dist_loss = 0.0, 0.0, 0.0
+        fg_bond_from_atom_loss, fg_bond_from_bond_loss, fg_bond_dist_loss = 0.0, 0.0, 0.0
+
+        if preds["av_task"][0] is not None:
+            av_atom_loss = av_task_loss(preds['av_task'][0], labels["av_task"])
+            fg_atom_from_atom_loss = fg_task_loss(
+                preds["fg_task"]["atom_from_atom"], labels["fg_task"])
+
+        if preds["av_task"][1] is not None:
+            av_bond_loss = av_task_loss(preds['av_task'][1], labels["av_task"])
+            fg_atom_from_bond_loss = fg_task_loss(
+                preds["fg_task"]["atom_from_bond"], labels["fg_task"])
+
+        if preds["bv_task"][0] is not None:
+            bv_atom_loss = av_task_loss(preds['bv_task'][0], labels["bv_task"])
+            fg_bond_from_atom_loss = fg_task_loss(
+                preds["fg_task"]["bond_from_atom"], labels["fg_task"])
+
+        if preds["bv_task"][1] is not None:
+            bv_bond_loss = av_task_loss(preds['bv_task'][1], labels["bv_task"])
+            fg_bond_from_bond_loss = fg_task_loss(
+                preds["fg_task"]["bond_from_bond"], labels["fg_task"])
+
+        if preds["av_task"][0] is not None and preds["av_task"][1] is not None:
+            av_dist_loss = av_task_dist_loss(preds['av_task'][0],
+                                             preds['av_task'][1])
+            fg_atom_dist_loss = fg_task_dist_loss(
+                sigmoid(preds["fg_task"]["atom_from_atom"]),
+                sigmoid(preds["fg_task"]["atom_from_bond"]))
+
+        if preds["bv_task"][0] is not None and preds["bv_task"][1] is not None:
+            bv_dist_loss = av_task_dist_loss(preds['bv_task'][0],
+                                             preds['bv_task'][1])
+            fg_bond_dist_loss = fg_task_dist_loss(
+                sigmoid(preds["fg_task"]["bond_from_atom"]),
+                sigmoid(preds["fg_task"]["bond_from_bond"]))
+
+        av_loss = av_atom_loss + av_bond_loss
+        bv_loss = bv_atom_loss + bv_bond_loss
+        fg_atom_loss = fg_atom_from_atom_loss + fg_atom_from_bond_loss
+        fg_bond_loss = fg_bond_from_atom_loss + fg_bond_from_bond_loss
+
+        fg_loss = fg_atom_loss + fg_bond_loss
+        fg_dist_loss = fg_atom_dist_loss + fg_bond_dist_loss
+
+        overall_loss = av_loss + bv_loss + fg_loss + dist_coff * av_dist_loss + \
+                       dist_coff * bv_dist_loss + fg_dist_loss
+
+        return overall_loss
+
+    @staticmethod
+    def atom_random_mask(atom_vocab, smiles_batch: List[str]):
+        """
+        Parameters
+        ----------
+        atom_vocab: grover.data.MolVocab
+            atom vocabulary
+        smiles_batch: List[str]
+            a list of smiles string
```

Comment:
Can you add a `Returns` field and a description of the purpose of this method to the docstring?